contracts/libraries/CircleCCTPAdapter.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../external/interfaces/CCTPInterfaces.sol";
+
+abstract contract CircleCCTPAdapter {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice The domain ID that CCTP will transfer funds to.
+     * @dev This identifier is assigned by Circle and is not related to a chain ID.
+     * @dev Official domain list can be found here: https://developers.circle.com/stablecoins/docs/supported-domains
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint32 public immutable recipientCircleDomainId;
+
+    /**
+     * @notice The official USDC contract address on this chain.
+     * @dev Posted officially here: https://developers.circle.com/stablecoins/docs/usdc-on-main-networks
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable usdcToken;
+
+    /**
+     * @notice The official Circle CCTP token bridge contract endpoint.
+     * @dev Posted officially here: https://developers.circle.com/stablecoins/docs/evm-smart-contracts
+     */
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ITokenMessenger public immutable cctpTokenMessenger;
+
+    /**
+     * @notice intiailizes the CircleCCTPAdapter contract.
+     * @param _usdcToken USDC address on the current chain.
+     * @param _cctpTokenMessenger TokenMessenger contract to bridge via CCTP. If the zero address is passed, CCTP bridging will be disabled.
+     * @param _recipientCircleDomainId The domain ID that CCTP will transfer funds to.
+     */
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        IERC20 _usdcToken,
+        ITokenMessenger _cctpTokenMessenger,
+        uint32 _recipientCircleDomainId
+    ) {
+        usdcToken = _usdcToken;
+        cctpTokenMessenger = _cctpTokenMessenger;
+        recipientCircleDomainId = _recipientCircleDomainId;
+    }
+
+    /**
+     * @notice converts address to bytes32 (alignment preserving cast.)
+     * @param addr the address to convert to bytes32
+     * @dev Sourced from the official CCTP repo: https://github.com/walkerq/evm-cctp-contracts/blob/139d8d0ce3b5531d3c7ec284f89d946dfb720016/src/messages/Message.sol#L142C1-L148C6
+     */
+    function _addressToBytes32(address addr) internal pure returns (bytes32) {
+        return bytes32(uint256(uint160(addr)));
+    }
+
+    /**
+     * @notice Returns whether or not the CCTP bridge is enabled.
+     */
+    function _isCCTPEnabled() private view returns (bool) {
+        return address(cctpTokenMessenger) != address(0) && address(usdcToken) != address(0);
```

Comment:
Prefer only needing to check `address(cctpTokenMessenger) != address(0)` to reduce gas cost. This then assumes we'll always set the `usdcToken` address correctly but we can set `cctpTokenMessenger` to 0x0 in order to "disable" this adapter

Reply:
Good call: https://github.com/across-protocol/contracts-v2/pull/391/commits/953e2915227dcfa528541d39e9930021ae8c86f9